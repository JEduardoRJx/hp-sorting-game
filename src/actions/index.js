export const setCharacters = characters => ({
  type: 'SET_CHAR',
  characters
})

export const setGameCharacters = allCharacters => ({
  type: 'SET_GAME_CHAR',
  allCharacters
})

export const isLoading = bool => ({
  type: 'IS_LOADING',
  bool
})

export const setHouse = house => ({
  type: 'SET_HOUSE',
  house
})