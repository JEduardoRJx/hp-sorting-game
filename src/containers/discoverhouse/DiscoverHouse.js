import React, { Component } from 'react';
import './DiscoverHouse.scss';
import PlayerInfo from '../../components/playerinfo/PlayerInfo'

export class DiscoverHouse extends Component {
  constructor() {
    super();
    this.state = {

    }
  }

  render() {
    return (
      <section className='discover-house-section'>
        <PlayerInfo />
        Discover House section
      </section>
    )
  }
}

export default DiscoverHouse;