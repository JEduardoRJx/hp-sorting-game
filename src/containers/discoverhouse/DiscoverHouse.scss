$hw-dark: #4c3638;
$hw-light: #D3A481;
$sl-dark: #276F45;
$sl-light: #A8B0BF;
$gryf-dark: #702737;
$gryf-light: #DAA520;
$rav-dark: #274F70;
$rav-light: #C98C4E;
$huff-dark: #141414;
$huff-light: #D1B738;

.discover-house-section {
  height: 90vh;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  justify-items: center;
}

.house-form {
  grid-column: 2/4;
  display: flex;
  width: 100%;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
}

.name-input {
  height: 30px;
  width: 400px;
  font-size: 2em;
  padding: 5px;
  text-align: center;
  border: none;
  border-radius: 5px;
  outline: none;
}

.house-label {
  font-size: 2em;
  color: $hw-light;
  &.gryffindor {
    color: $gryf-light;
  }
  &.slytherin {
    color: $sl-light;
  }
  &.hufflepuff {
    color: $huff-light;
  }
  &.ravenclaw {
    color: $rav-light;
  }
}

.witchwizard-btn-wrapper {
  display: flex;
  justify-content: space-around;
  width: 75%;
}

.witchwizard-btn {
  width: 200px;
  font-size: 2em;
}

.wand-btn-wrapper {
  display: grid;
  grid-template-columns: 1fr 1fr;
  height: 200px;
  width: 100%;
  align-items: center;
  justify-items: center;
}

.wand-btn {
  width: 350px;
  height: 60px;
  font-size: 1.5em;
}

.your-house-btn {
  width: 50%;
  font-size: 2.5em;
}

.witchwizard-btn,
.wand-btn,
.your-house-btn {
  border-radius: 30px;
  border: none;
  color: $hw-light;
  background-color: $hw-dark;
  border: 3px solid $hw-light;
  outline: none;
  &:hover {
    color: $hw-dark;
    background-color: $hw-light;
    border: 3px solid $hw-dark;
  }
  &.gryffindor {
    color: $gryf-light;
    background-color: $gryf-dark;
    border: 3px solid $gryf-light;
    &:hover {
      color: $gryf-dark;
      background-color: $gryf-light;
      border: 3px solid $gryf-dark;
    }
  }
  &.slytherin {
    color: $sl-light;
    background-color: $sl-dark;
    border: 3px solid $sl-light;
    &:hover {
      color: $sl-dark;
      background-color: $sl-light;
      border: 3px solid $sl-dark;
    }
  }
  &.hufflepuff {
    color: $huff-light;
    background-color: $huff-dark;
    border: 3px solid $huff-light;
    &:hover {
      color: $huff-dark;
      background-color: $huff-light;
      border: 3px solid $huff-dark;
    }
  }
  &.ravenclaw {
    color: $rav-light;
    background-color: $rav-dark;
    border: 3px solid $rav-light;
    &:hover {
      color: $rav-dark;
      background-color: $rav-light;
      border: 3px solid $rav-dark;
    }
  }
}

.house-modal {
  display: none;
  grid-column: 2/4;
  flex-direction: column;
  width: 100%;
  text-align: center;
  align-items: center;
  justify-content: space-around;
}

.house-home-btn  {
  font-size: 2em;
  width: 50%;
  text-decoration: none;
  background-color: $hw-dark;
  color: $hw-light;
  border-radius: 3px;
  padding: 5px;
  border: 3px solid $hw-light;
  &:hover {
    background-color: $hw-light;
    color: $hw-dark;
  }
  &.gryffindor {
    background-color: $gryf-dark;
    color: $gryf-light;
    border: 3px solid $gryf-light;
    &:hover {
      background-color: $gryf-light;
      color: $gryf-dark;
    }
  }
  &.slytherin {
    background-color: $sl-dark;
    color: $sl-light;
    border: 3px solid $sl-light;
    &:hover {
      background-color: $sl-light;
      color: $sl-dark;
    }
  }
  &.hufflepuff {
    background-color: $huff-dark;
    color: $huff-light;
    border: 3px solid $huff-light;
    &:hover {
      background-color: $huff-light;
      color: $huff-dark;
    }
  }
  &.ravenclaw {
    background-color: $rav-dark;
    color: $rav-light;
    border: 3px solid $rav-light;
    &:hover {
      background-color: $rav-light;
      color: $rav-dark;
    }
  }
}

.house-h1 {
  font-size: 10em;
  font-family: harry_pregular;
  &.gryffindor {
    color: $gryf-light;
  }
  &.slytherin {
    color: $sl-light;
  }
  &.hufflepuff {
    color: $huff-light;
  }
  &.ravenclaw {
    color: $rav-light;
  }
}