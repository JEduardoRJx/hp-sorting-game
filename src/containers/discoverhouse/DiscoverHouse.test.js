import React from 'react';
import { shallow } from 'enzyme';
import { DiscoverHouse, mapStateToProps, mapDispatchToProps } from './DiscoverHouse';
import { setUser } from '../../actions';

describe('DiscoverHouseContainer', () => {

  describe('DiscoverHouse component', () => {
    let wrapper;
    const mockSetUser = jest.fn();
    const mockHouse = 'Slytherin'
    const mockUser = {
      name: 'name',
      wizardkind: 'wizard',
      wand: 'wand',
      house: 'slytherin'
    }
    
    beforeEach(() => {
      wrapper = shallow (<DiscoverHouse 
        setUser={mockSetUser} 
        house={mockHouse}
        user={mockUser}
      />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should update state when handleName is called',() => {
      const mockEvent = {
        target: {
          value: 'name'
        }
      }
      const expected = 'name'
      wrapper.instance().handleName(mockEvent)
      expect(wrapper.state('name')).toEqual(expected);
    });

    it('should update state when handleWizardKindAndWand is called',() => {
      const mockEvent = {
        target: {
          name: 'wand',
          value: 'magical wand'
        }
      }
      const expected = 'magical wand'
      wrapper.instance().handleWizardkindAndWand(mockEvent)
      expect(wrapper.state('wand')).toEqual(expected);
    });

    
  });

  describe('mapStateToProps', () => {

  });

  describe('mapDispatchToProps', () => {
    
  })
})