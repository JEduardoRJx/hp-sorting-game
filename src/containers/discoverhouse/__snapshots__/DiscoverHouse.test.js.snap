// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscoverHouseContainer DiscoverHouse component should match snapshot 1`] = `
<section
  className="discover-house-section"
>
  <Connect(PlayerInfo) />
  <div
    className="house-modal"
  >
    <h1
      className="house-h1 slytherin"
    >
      SLYTHERIN
    </h1>
    <NavLink
      className="house-home-btn slytherin"
      to="/"
    >
      Back to the Great Hall
    </NavLink>
  </div>
  <form
    className="house-form"
  >
    <input
      className="name-input"
      onChange={[Function]}
      placeholder="Enter Name"
      type="text"
    />
    <h2
      className="house-label slytherin"
    >
      Witch or Wizard
    </h2>
    <div
      className="witchwizard-btn-wrapper"
    >
      <button
        className="witchwizard-btn slytherin"
        name="wizardkind"
        onClick={[Function]}
        type="button"
        value="Witch"
      >
        Witch
      </button>
      <button
        className="witchwizard-btn slytherin"
        name="wizardkind"
        onClick={[Function]}
        type="button"
        value="Wizard"
      >
        Wizard
      </button>
    </div>
    <h2
      className="house-label slytherin"
    >
      Select your Wand
    </h2>
    <div
      className="wand-btn-wrapper"
    >
      <button
        className="wand-btn slytherin"
        name="wand"
        onClick={[Function]}
        type="button"
        value="Elder, 15\\", thestral tail hair"
      >
        Elder, 15", thestral tail hair
      </button>
      <button
        className="wand-btn slytherin"
        name="wand"
        onClick={[Function]}
        type="button"
        value="Holly, 11\\", phoenix feather"
      >
        Holly, 11", phoenix feather
      </button>
      <button
        className="wand-btn slytherin"
        name="wand"
        onClick={[Function]}
        type="button"
        value="Vine wood, 10 3/4\\", dragon heartstring"
      >
        Vine wood, 10 3/4", dragon heartstring
      </button>
      <button
        className="wand-btn slytherin"
        name="wand"
        onClick={[Function]}
        type="button"
        value="Fir, 9 1/2\\", dragon heartstring"
      >
        Fir, 9 1/2", dragon heartstring
      </button>
    </div>
  </form>
</section>
`;
